### ISSUE TITLE
Proposal: Introduce `meta` object to Dispatcher#dispatch(payload, meta)

### ISSUE THREAD

**Issue by azu** [almin/almin#61]
## Motivation

Currently, `Dispatcher#dispatch(payload)` can dispatch some payload.
But, The receiver of payload can't know  that is the `payload` come from?

The feature is needed to implement`dispatch` logging correctly.

- https://github.com/azu/almin-logger want to know place of payload origin!

## Propose

Introduce `Dispatcher#dispatch(payload, meta)`
`dispatch`'s second argument is contextual object.
This `meta` object contains: 

- `useCase`: current UseCase
- `parentUseCase` : parent UseCase
- <del>`error`?: error object when `throwError`</del> To be payload.
- <del>`args`?:  arguments when `willExecute`</del> To be payload.
- <del>`results`? : results when `didExecute` . maybe rename</del> To be payload.
- `timeStamp`: timeStamp

`meta` means that this is created by almin. ( Can we allow the user to use `meta`?)

## Breaking Change

They should be changed in the proposal:

```js
Context#onDispatch((payload, meta) => {});
Context#onWillExecuteEachUseCase((payload, meta) => { });
Context#onCompleteEachUseCase((payload, meta) => { });
Context#onDidExecuteEachUseCase((payload, meta) => { });
Context#onErrorDispatch((payload, meta) => {});
```

- `payload`: `DispatcherPayload`
    - `type`: `*` - maybe String or Symbol
    - `error`?: error object when `throwError`
    - `args`?:  arguments when `willExecute`
    - `results`? : results when `didExecute` . maybe rename. require more better name
- `meta`: `Object`
    - `useCase`: current UseCase
    - `parentUseCase` : parent UseCase
    - `timeStamp`: timeStamp

## 📝 Note: vs. DOM [Event](https://developer.mozilla.org/en-US/docs/Web/API/Event "Event - Web APIs | MDN")

DOM [Event](https://developer.mozilla.org/en-US/docs/Web/API/Event "Event - Web APIs | MDN") has all property in own `event` object.
`event.type`, `event.target`, `event.timeStamp` and so....

The difference between DOM Event and Almin payload(`DispatcherPayload`).

DOM Event object is not must that added Custom property like 

```js
event.customeProp = 42
```

But, Almin's payload must that added Custom property like 

```js
{ type : "custom" , customProps: 42 }
```

Separated `meta` object is required to grow ecosystem of Almin.
`meta` object is always attached by Almin core system.
In other word, the user can't attached `meta` obejct.

**Comment by azu** [comment]
📝 UseCase will need `id`.

**Comment by azu** [comment]
Merged #62
